lib/java/com/google/android/material/lists/item/Visual.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2020 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.material.lists.item;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.Gravity;
+import android.widget.LinearLayout;
+
+import androidx.annotation.DimenRes;
+import androidx.annotation.Dimension;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.google.android.material.R;
+import com.google.android.material.imageview.ShapeableImageView;
+import com.google.android.material.internal.ViewUtils;
+import com.google.android.material.shape.CornerFamily;
+import com.google.android.material.shape.ShapeAppearanceModel;
+
+public class Visual extends ShapeableImageView implements TotalLinesListener {
+
+  protected Visual(Context context) {
+    super(context);
+  }
+
+  public Visual(Context context, @Nullable AttributeSet attrs) {
+    super(context, attrs);
+  }
+
+  public Visual(Context context, @Nullable AttributeSet attrs, int defStyle) {
+    super(context, attrs, defStyle);
+  }
+
+  public static final int NONE = 0;
```

Comment:
Use [`IntDef`](https://developer.android.com/reference/androidx/annotation/IntDef) for these?

Reply:
I was considering using an enum for the task, however it adds on an extra portion to the call site: `MyClass.EnumType.NONE` instead of `MyClass.NONE`. IntDef seems like a good idea to annotate the "enum"

Reply:
applied